test/functional/page_objects/common_page.ts

```diff
@@ -164,116 +202,84 @@ export function CommonPageProvider({ getService, getPageObjects }) {
-              // (from debug output). Some other tabs may also be long. But a rather simple configured
-              // visualization is about 1000 chars long. So at least we catch that case.
-
-              // Browsers don't show the ':port' if it's 80 or 443 so we have to
-              // remove that part so we can get a match in the tests.
-              const navSuccessful = new RegExp(appUrl.replace(':80/', '/').replace(':443/', '/')
-                + '.{0,' + maxAdditionalLengthOnNavUrl + '}$')
-                .test(currentUrl);
-
-              if (!navSuccessful) {
-                const msg = 'App failed to load: ' + appName +
-                  ' in ' + defaultFindTimeout + 'ms' +
-                  ' appUrl = ' + appUrl +
-                  ' currentUrl = ' + currentUrl;
-                log.debug(msg);
-                throw new Error(msg);
-              }
-
-              return currentUrl;
-            });
+          await CommonPage.navigateToUrlAndHandleAlert(appUrl, shouldAcceptAlert);
+          await this.sleep(700);
+          log.debug('returned from get, calling refresh');
+          await browser.refresh();
+          let currentUrl = shouldLoginIfPrompted
+            ? await this.loginIfPrompted(appUrl)
+            : await browser.getCurrentUrl();
+
+          if (currentUrl.includes('app/kibana')) {
+            await testSubjects.find('kibanaChrome');
+          }
+
+          currentUrl = (await browser.getCurrentUrl()).replace(/\/\/\w+:\w+@/, '//');
+          const maxAdditionalLengthOnNavUrl = 230;
+
+          // On several test failures at the end of the TileMap test we try to navigate back to
+          // Visualize so we can create the next Vertical Bar Chart, but we can see from the
+          // logging and the screenshot that it's still on the TileMap page. Why didn't the "get"
+          // with a new timestamped URL go? I thought that sleep(700) between the get and the
+          // refresh would solve the problem but didn't seem to always work.
+          // So this hack fails the navSuccessful check if the currentUrl doesn't match the
+          // appUrl plus up to 230 other chars.
+          // Navigating to Settings when there is a default index pattern has a URL length of 196
+          // (from debug output). Some other tabs may also be long. But a rather simple configured
+          // visualization is about 1000 chars long. So at least we catch that case.
+
+          // Browsers don't show the ':port' if it's 80 or 443 so we have to
+          // remove that part so we can get a match in the tests.
+          const navSuccessful = new RegExp(
+            appUrl.replace(':80/', '/').replace(':443/', '/') +
+              '.{0,' +
+              maxAdditionalLengthOnNavUrl +
+              '}$'
+          ).test(currentUrl);
+
+          if (!navSuccessful) {
+            const msg =
+              'App failed to load: ' +
```

Comment:
Is it required to break this up into so many lines :-(

![image](https://user-images.githubusercontent.com/13542669/68970576-3483a200-07ad-11ea-8c5a-22e1ef35f287.png)

Reply:
A template literal would make this much more readable.

Reply:
Agree, I will fix it :)

Reply:
fixed in a362ca31892bfb0d0b963ac8d20dd0eddb1bb2a6